reco_utils/common/timer.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from timeit import default_timer
+from datetime import timedelta
+
+
+class Timer(object):
+    """Timer class.
+    Original code: https://github.com/miguelgfierro/codebase
+    
+    Examples:
+        >>> big_num = 1000
+        >>> t = Timer()
+        >>> t.start()
+        >>> r = 0
+        >>> a = [r+i for i in range(big_num)]
+        >>> t.stop()
+        >>> t.interval < 1
+        True
```

Comment:
quick question. if a user calls `t.interval` without calling `t.stop()`, it will give wrong value. will that okay? few suggestions:
* reset the interval whenever `start()` get called
* make the interval as a private var `_interval` and `stop()` returns it. e.g. `interval = my_timer.stop()` this will resolve the issue of the user-mistake I mentioned.
* maybe add `lap()`? which return the current interval and reset & restart timer internally...do we need this? not sure. if we measure something in loops.. maybe.

Reply:
good catch. I'll think about how to fix it, not sure how seting interval = my_timer.stop() will work with the contextual format, maybe it is easier to have a property and then raise an exception if the timer is not stopped. 

for your other questions, when start is called, interval is resetted. And lap is the same as interval

Reply:
I changed this behavior, now if you call interval before stopping you get an exception.